### PR TITLE
Add missing completions in order-by-clause context

### LINKED ISSUE #33895: Order direction keywords are not suggested in order-clause context
**Description:**
```ballerina
function findToptenCountriesByAthletes() returns error? {
    map<int> athletesByCountry = check findNumOfAthletesBycountry();
    var result = from [string,int] entry in athletesByCountry.entries() 
                     order by entry[1] <cursor>
}
```

The completion list at the depicted cursor position does not contain the `ascending` and `descending` keywords.
This issue seems to be with the resolved completion context.

**Steps to reproduce:**
Mentioned above

### PR DESCRIPTION
**Pull request by malinthar**
## Purpose
Fix routing in indexed expression context and refactor OrderByClause Context. 

Fixes #33895 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [x] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/IndexedExpressionNodeContext.java [hunk 1]
```diff
@@ -55,7 +56,12 @@ public IndexedExpressionNodeContext() {
...
         }
         this.sort(context, node, completionItems);
-        
+
         return completionItems;
     }
+
+    @Override
+    public boolean onPreValidation(BallerinaCompletionContext context, IndexedExpressionNode node) {
+        return context.getNodeAtCursor().lineRange().endLine().offset() >= context.getCursorPosition().getCharacter();
```
Comment (nadeeshaan) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/IndexedExpressionNodeContext.java:65]: can't we use the `context.getCursorPositionInTree` and the text offsets for this comparison?

**Review by nadeeshaan — CHANGES_REQUESTED** [review]

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/IndexedExpressionNodeContext.java (hunk 1 above)
Comment (malinthar) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/IndexedExpressionNodeContext.java:65]: Yes we can. Will do

Comment (malinthar) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/IndexedExpressionNodeContext.java]: Done

**Review by IMS94 — APPROVED** [review]
LGTM!

**Review by nadeeshaan — APPROVED** [review]
LGTM

**Pull request merged**